src/main/java/com/google/javascript/gents/GentsCodeGenerator.java

```diff
@@ -159,6 +169,14 @@ boolean maybeOverrideCodeGen(Node n) {
           return true;
         }
         return false;
+      case FUNCTION_TYPE:
```

Comment:
I don't understand why the code that adds "(" and the one that adds ")" have to be so far apart. How can you convince me that we produce balanced parens?

Reply:
The way `maybeOverrideCodeGen` works is that if it returns true then no further code generation on the node is needed. We cannot use that for a `FUNCTION_TYPE` node otherwise we have to duplicate all the code gen logic for `FUNCTION_TYPE` which contains many invocations to private APIs.
So we have to rely on the default code gen for `FUNCTION_TYPE`, i.e. `maybeOverrideCodeGen` must return false for `FUNCTION_TYPE`. Therefore, to add a pair of `()` we need to add `(` before the code gen and `)` after the code gen.

I added some comments explaining this.